### DISCUSSION TITLE
How to create a reference for an admonition

### DISCUSSION THREAD

**Discussion by andrewnicols** [facebook/docusaurus#7146] [Q&A]
I'm trying to find a way to create a bookmark link for an admonition so that I can essentially provide more detail for terminology, however the `{#my-explicit-id}` syntax documented at https://docusaurus.io/docs/markdown-features/headings#explicit-ids seems only be for headings.

Given the admonition _does_ include a header, should it not also be possible to link to it?

**Comment by slorber** [comment]
I don't understand sorry. Do you have any example in the wild (on any doc site) using a setup similar to what you are trying to do?

> Given the admonition does include a header, should it not also be possible to link to it?

I'm not sure headers in admonitions is a good doc / HTML practice, but it should work and you should be able to link to any heading (unless you put them in collapsible/hideable items like details or tabs)

**Reply by Josh-Cena** [reply]
@slorber The admonition compiles to an `h5` heading. Complained here: https://github.com/elviswolcott/remark-admonitions/issues/26 But because it doesn't have an `id`, it's not actually an anchor.

**Reply by andrewnicols** [reply]
Sorry, I don't currently have an example but I'll try and find one tomorrow.

I want to be able to provide an anchor on an admonition to allow cross-referencing to it within the doc.

Since there is no jargon or terminology plugin for Docusaurs, we're currently looking at using an `info` admonition to describe more information about a key word, and then linking to that keyword, but right now we can't actually link to the admonition.

**Reply by slorber** [reply]
Using a headng in an admonition seems to work and produce an anchor

<img width="1282" alt="CleanShot 2022-04-13 at 19 00 11@2x" src="https://user-images.githubusercontent.com/749374/163232420-f3e949f0-806d-40ec-90d0-e595a4999118.png">

(but no TOC entry)

Is the goal to have a syntax like this?

```
:::info Keyword {#keyword}

some text

:::
```

**Reply by homotechsual** [reply]
> Is the goal to have a syntax like this?
```mdx
:::info Keyword {#keyword}
some text
:::
```

This would be ace.

**Reply by yasutakeyohei** [reply]
I've written a blog post on how to achieve this with swizzling, remark, and rehype. Hope you find it useful.

The syntax is like:

```md
:::info ### title_here

some text

:::
```

https://yasutakeyohei.com/docs/blog/2024/01/27/docusaurus-admonition-heading-toc/

**Reply by levino** [reply]
Is this still not possible without heavy lifting?

**Reply by slorber** [reply]
This issue is old and we haven't implemeted anything 

Note that with MDX, you can use JSX in admonitions so it's technically possible to add anchors anywhere

```mdx
:::note

<a id="my-linking-id" />

My text

:::
```

**Reply by levino** [reply]
It kinda works but I think in such cases, there is no offset being calculated and the content is hidden (partially at least) behind the navbar. I added a heading before the admonition and scrolling to that works much better.
